otc-swap-predicate/src/main.sw

```diff
@@ -0,0 +1,63 @@
+predicate;
+
+use std::{
+    inputs::{
+        input_count,
+        input_owner,
+    },
+    outputs::{
+        Output,
+        output_amount,
+        output_pointer,
+        output_type,
+    },
+};
+
+// TO DO : Remove once __gtf getters implemented in std-lib
+const GTF_OUTPUT_COIN_TO = 0x202;
+const GTF_OUTPUT_COIN_ASSET_ID = 0x204;
+
+/// Order / OTC swap Predicate
+fn main() -> bool {
+    // Order conditions: These are set in Forc.toml
+    // The spending transaction must have an output that sends `ask_amount` of `ask_token` to `receiver`
+    // Conversion to ContractId and Address types will be unnecessary once
+    // https://github.com/FuelLabs/sway/issues/2647 is fixed
+    let ask_token: ContractId = ContractId {
+        value: ask_token_config,
+    };
+    let receiver = ~Address::from(receiver_config);
+
+    // Check if the transaction contains a single input coin from the receiver, to cancel their own order
+    // Note that the predicate is necessarily one of the inputs, so the other must be the coin input.
+    if (input_count() == 2u8) {
+        let owner = match input_owner(0) {
+            Option::Some(owner) => owner,
+            _ => input_owner(1).unwrap(),
+        };
+
+        if (owner == receiver) {
+            return true;
+        };
+    };
```

Comment:
```suggestion
    if input_count() == 2u8 {
        let owner = match input_owner(0) {
            Option::Some(owner) => owner,
            _ => input_owner(1).unwrap(),
        };

        if owner == receiver {
            return true;
        };
    };
```

Not much going on here so I don't believe that wrapping with parenthesis helps readability.

Noting that the `2u8` is padded to `u64` so I'm unsure what actually happens here. Hopefully  the tests are sufficient and document this in case this gets fixed in the VM in the future

Reply:
resolved in [c0a5fc1](https://github.com/FuelLabs/sway-applications/pull/142/commits/c0a5fc1c00ce18d3fafca269cb4e0fac78b2837d)

`input_count()` returns a `u8` so this annotation avoids the "lose precision" compiler warning